features/cellular/framework/AT/ATHandler.cpp

```diff
@@ -515,15 +515,15 @@ ssize_t ATHandler::read(char *buf, size_t size, bool read_even_stop_tag, bool he
            pbuf[buf_idx] = c;
         } else {
             if (read_idx % 2 == 0) {
-                upper = hex_str_to_int(&c, 1);
+                upper = hex_str_to_int((const char*)&c, 1);
```

Comment:
That cast is not legal - you're going to need to actually have a `char` to pass this function.

Suggest the simplest way might not be to remember `upper` but to remember the previous character.

     char hexbuf[2];

     hexbuf[read_idx % 2] = c;
     if (read_idx % 2 == 1) {
        hex_str_to_char_str(hexbuf, 2, pbuf+buf_idx);
     }